src/EditorFeatures/CSharpTest/ConvertAnonymousTypeToTuple/ConvertAnonymousTypeToTupleTests.cs

```diff
@@ -0,0 +1,328 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.ConvertAnonymousTypeToTuple;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertAnonymousTypeToTuple
+{
+    public partial class ConvertAnonymousTypeToTupleTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
```

Comment:
>{ [](start = 4, length = 1)

Would you mind testing in various containers? (local function, lambda)

Reply:
Will do.

Reply:
Maybe also test on incomplete code `new { a = , b = }`?

---
In reply to: [197975162](https://github.com/dotnet/roslyn/pull/27099#discussion_r197975162) [](ancestors = 197975162)

Reply:
tests added (for both C# and VB).